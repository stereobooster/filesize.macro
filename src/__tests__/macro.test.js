const path = require("path");
const pluginTester = require("babel-plugin-tester");
const plugin = require("babel-plugin-macros");
const prettier = require("prettier");

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename,
  },
  formatResult(result) {
    return prettier.format(result, { trailingComma: "es5" });
  },
  tests: {
    "no usage": `import filesize from '../macro'`,
    "correct usage": `
      import filesize from '../macro';

      const preview = filesize('./fixtures/jonathan-daniels-416786-unsplash.jpg');
    `,
    // "correct usage, wrong file": `
    //   import filesize from '../macro';

    //   const preview = filesize('./fixtures/readme.md');
    // `,
  },
});
