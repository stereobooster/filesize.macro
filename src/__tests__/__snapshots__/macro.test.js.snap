// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros correct usage: correct usage 1`] = `
"
import filesize from '../macro';

const preview = filesize('./fixtures/jonathan-daniels-416786-unsplash.jpg');

      ↓ ↓ ↓ ↓ ↓ ↓

const preview = 709289;

"
`;

exports[`macros no usage: no usage 1`] = `
"
import filesize from '../macro'

      ↓ ↓ ↓ ↓ ↓ ↓


"
`;
